File: src/main/java/uk/co/threebugs/mochiwhattotrade3/accountsetupgroups/AccountSetupGroupsMapper.java
package uk.co.threebugs.mochiwhattotrade3.accountsetupgroups;

import org.mapstruct.Mapper;

import java.nio.file.Path;

@Mapper(componentModel = "spring")
public interface AccountSetupGroupsMapper {
    AccountSetupGroups toEntity(AccountSetupGroupsDto accountSetupGroupsDto);

    AccountSetupGroupsDto toDto(AccountSetupGroups accountSetupGroups);

    default Path toPath(String path) {
        return Path.of(path);
    }

    default String toString(Path path) {
        return path.toString();
    }
}

File: src/main/kotlin/uk/co/threebugs/mochiwhattotrade3/accountsetupgroups/AccountSetupGroupsMapper.kt
package uk.co.threebugs.mochiwhattotrade3.accountsetupgroups

import org.mapstruct.Mapper
import java.nio.file.Path

@Mapper(componentModel = "spring")
abstract class AccountSetupGroupsMapper {
    abstract fun toEntity(accountSetupGroupsDto: AccountSetupGroupsDto): AccountSetupGroups
    abstract fun toDto(accountSetupGroups: AccountSetupGroups): AccountSetupGroupsDto
    fun toPath(path: String?): Path? {
        return path?.let { Path.of(it) }
    }

    fun toString(path: Path?): String? {
        return path?.toString()
    }
}
